### ISSUE TITLE
corner-radius-rules not working

### ISSUE THREAD

**Issue by sawntoe** [yshui/compton#1170]
### Platform
Arch Linux 6.7.1-arch1-1 

### GPU, drivers, and screen setup

 Setup: X11 w/ i3, 1 screen
GPU: Intel TigerLake-LP GT2 [Iris Xe Graphics]
xf86-video-intel 1:2.99.917+923+gb74b67f0-1

```
$ glxinfo -B
name of display: :0
display: :0  screen: 0
direct rendering: Yes
Extended renderer info (GLX_MESA_query_renderer):
    Vendor: Intel (0x8086)
    Device: Mesa Intel(R) Xe Graphics (TGL GT2) (0x9a49)
    Version: 23.3.3
    Accelerated: yes
    Video memory: 15546MB
    Unified memory: yes
    Preferred profile: core (0x1)
    Max core profile version: 4.6
    Max compat profile version: 4.6
    Max GLES1 profile version: 1.1
    Max GLES[23] profile version: 3.2
OpenGL vendor string: Intel
OpenGL renderer string: Mesa Intel(R) Xe Graphics (TGL GT2)
OpenGL core profile version string: 4.6 (Core Profile) Mesa 23.3.3-arch1.1
OpenGL core profile shading language version string: 4.60
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile

OpenGL version string: 4.6 (Compatibility Profile) Mesa 23.3.3-arch1.1
OpenGL shading language version string: 4.60
OpenGL context flags: (none)
OpenGL profile mask: compatibility profile

OpenGL ES profile version string: OpenGL ES 3.2 Mesa 23.3.3-arch1.1
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.20
```

### Environment

i3-git

### picom version vgit-702e3

<details>
<summary>Diagnostics</summary>

```
**Version:** vgit-702e3

### Extensions:

* Shape: Yes
* RandR: Yes
* Present: Present

### Misc:

* Use Overlay: No
  (Another compositor is already running)
* Config file used: /home/sawntoe/.config/picom/picom.conf

### Drivers (inaccurate):

Intel

### Backend: glx

* Driver vendors:
 * GLX: Mesa Project and SGI
 * GL: Intel
* GL renderer: Mesa Intel(R) Xe Graphics (TGL GT2)
* Accelerated: 1

### Backend: egl

* Driver vendors:
 * EGL: Mesa Project
 * EGL driver: iris
 * GL: Intel
* GL renderer: Mesa Intel(R) Xe Graphics (TGL GT2)
```
</details>

### Configuration:
<details>
    <summary>Configuration file</summary>

```

backend = "glx";
glx-no-stencil = true;
glx-copy-from-front = false;

# Opacity
active-opacity = 1;
inactive-opacity = 1;
frame-opacity = 1;
inactive-opacity-override = false;
blur-background = true;

blur-method = "dual_kawase";
blur-strength = 6;


# Fading
fading = true;
fade-delta = 4;
no-fading-openclose = false;

fade-exclude = [ ];

# Other
mark-wmwin-focused = true;
mark-ovredir-focused = true;
detect-rounded-corners = true;
detect-client-opacity = true;
refresh-rate = 0;
vsync = false;
dbe = false;
unredir-if-possible = false;
focus-exclude = [ ];
detect-transient = true;
detect-client-leader = true;

# Window type settings
wintypes:
{
		dock = {
				shadow = false;
		};
};


opacity-rule = [
# "80:class_g = 'Rofi'",
		"60:class_g = 'Rofi'",
		"90:class_g = 'discord'",
		"90:class_g = 'Subl'",
		"80:class_g = 'Microsoft Teams - Preview'"
];

blur-background-exclude = [ "window_type = 'dock'",
			   "window_type = 'desktop'", 
			   "_GTK_FRAME_EXTENTS@:c", 
			   "class_g = 'slop'",
			   "role = 'xborder'",
			   "class_g = 'GLava'",
			   "class_g = 'Florence'",
			   ];

xrender-sync-fence = true;
animations = false;
transition-length = 0;
transition-pow-x = 0;
inactive-dim = 0;

# corner-radius = 0;

# rounded-corners-exclude = [
#   "class_g = 'Polybar'",
#   "class_g = 'Dunst'"
# ];

corner-radius-rules = [
  "15:class_g = 'Polybar'",
  "7:class_g = 'Dunst'"
];
```
</details>

### Steps of reproduction

1. Run picom with `corner-radius-rules` configured

### Expected behavior

The corners of the windows specified would have their own specific corner radii

### Current Behavior

Default corner radius is applied to all windows

**Comment by yshui** [comment]
what does xprop show for the polybar window?

**Comment by absolutelynothelix** [comment]
it works but only when `corner-radius` is not `0`. i consider this a bug.

**Comment by dglava** [comment]
I am still getting this issue with Picom version 11.2 (on Arch Linux).

```
corner-radius = 0

corner-radius-rules = [
  "10:window_type = 'popup_menu'"
];
```

In this example I want to apply a radius of 10 only to popup menus. If I set `corner-radius = 0`, they're not having rounded corners. If I set the radius to even just 1 `corner-radius = 1`, the popup menus have a radius of 10, but so do all of the other windows have a radius of 1 now (not really visible, but still).
Am I doing something wrong?
